CRM/Civixero/Contact.php

```diff
@@ -190,7 +192,13 @@ public function push(array $params, int $limit = 10): bool {
         $locationTypeToSync = (int) Civi::settings()->get('xero_sync_location_type');
         $contact['email'] = $this->getPreferredEmail($locationTypeToSync, $record['contact_id']);
         $contact['phone'] = $this->getPreferredPhone($locationTypeToSync, $record['contact_id']);
-        $contact += $this->getPreferredAddress($locationTypeToSync, $record['contact_id']);
+        // Address is different to the other location fields because it has multiple fields.
+        // We might return NULL from getPreferredAddress which means "do not sync to Xero".
+        // That way we preserve any partial address that we might have in Xero and it will be synced next time it's pulled to Civi.
+        $contactAddress = $this->getPreferredAddress($locationTypeToSync, $record['contact_id']);
```

Comment:
This part seems like it makes sense to  bring over - I'm not quite sold on the return array - I'd be OK if it were more in line with a get api - ie an array of ['values'] => [5 => ['id' => 5']] or if we picked up a general discussion about apiv4 `Job` type api calls